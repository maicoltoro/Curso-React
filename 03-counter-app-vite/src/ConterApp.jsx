import { useState } from 'react'
import PropTypes  from "prop-types"

export const Tarea = ({valor}) =>{

    const [ counter, setCounter ] = useState( valor )

    const sumar = (event) =>{
        setCounter( (c) => c + 1 )
    }

    const restar = (event) =>{
        setCounter( (c) => c - 1 )
    }

    const Restaurar = (event) =>{
        setCounter( (c) => c = valor )
    }

    return (
        <>
            <h1> ConterApp </h1>
            <h2>{ counter }</h2>
            
            <button onClick={ (event) => sumar(event) }> +1  </button>
            <button onClick={ (event) => restar(event) }> -1  </button>
            <button onClick={ (event) => Restaurar(event) }> Reset </button>

        </>
    )
}

Tarea.protoType = {
    valor : PropTypes.number.isRequired
}