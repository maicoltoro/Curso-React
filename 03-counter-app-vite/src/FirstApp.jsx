import PropTypes  from "prop-types"

export const Componente = ({title , subTitle,name}) =>{

    return (
        <>
            <h1 data-testid="test-title" >{title}</h1>
            {/* <h1> {JSON.stringify(mensaje)} </h1> */}
            <p> {subTitle}</p>
            <p> {name}</p>
        </>
    )
}

Componente.protoType = {
    title : PropTypes.string.isRequired,
    subTitle : PropTypes.string.isRequired,
}

Componente.defaultProps = {
    subTitle : 'No hay subtitulo',
    name: 'maicol toro'
}