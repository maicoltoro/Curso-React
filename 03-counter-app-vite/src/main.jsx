import React from 'react'
import ReactDOM from 'react-dom/client'
import { Tarea } from './ConterApp'
import { Componente } from './FirstApp'
import {App} from './HelloWorldApp'

import './style.css'

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <Tarea valor={15} /> */}
        <Componente title="pruebas de react"></Componente>
    </React.StrictMode>
)