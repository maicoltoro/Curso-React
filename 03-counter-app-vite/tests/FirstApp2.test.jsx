import { render,screen } from "@testing-library/react"
import {Componente} from '../src/FirstApp'

describe('pruebas en FirstApp', () => { 
    
    const title = "prueba test"


    test('debes de hacer match con el snapshot', () => { 

        const {container} = render(<Componente title = {title}/>)
        expect (container).toMatchSnapshot();
    })

    test('debes de mostrar el mensaje', () => { 

        render(<Componente title = {title}/>)
        expect(screen.getByText(title)).toBeTruthy();
    })

    test('debes de mostrar el titulo en un h1', () => { 

        render(<Componente title = {title}/>)
        //console.log(screen.getByRole("heading",{level: 1}))
        expect(screen.getByRole('heading',{ level :1 }).innerHTML).toContain(title);
    })

 })