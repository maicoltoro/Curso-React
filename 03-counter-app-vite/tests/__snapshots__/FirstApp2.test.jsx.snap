// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstApp debes de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    prueba test
  </h1>
  <p>
     
    No hay subtitulo
  </p>
  <p>
     
    maicol toro
  </p>
</div>
`;
